### ISSUE TITLE
was7/8.5环境template.render(model, out)方法报错，weblogic环境没问题

### ISSUE THREAD

**Issue by PanShaobo** [httl/httl#210]
该问题并不固定出现，在重启应用时出现。
问题表现为，部署在was环境时，执行template.render(model, out)后台异常，返回结果为空。
异常：
java.lang.NullPointerException
    at java.io.Writer.write(Writer.java:121)
    at httl.spi.translators.templates.Template__ba_bee_vm_baseView_js_product_GBK_writer.doRenderWriter(Template__ba_bee_vm_baseView_js_product_GBK_writer.java)
    at httl.spi.translators.templates.CompiledTemplate.doRender(CompiledTemplate.java:149)
    at httl.spi.translators.templates.AbstractTemplate._render(AbstractTemplate.java:153)
    at httl.spi.translators.templates.AbstractTemplate.render(AbstractTemplate.java:142)

跟踪代码发现：静态变量为空
如：private static final char[] $TXT1 = httl.util.CharCache.getAndRemove("59");
导致： 
Writer方法中的该函数cbuf.length处报错
  public void write(char cbuf[]) throws IOException {
    write(cbuf, 0, cbuf.length);
    }
猜测：可能与ibm的jdk有关。

该问题在部署war后，多次（2次即可）重启应用后必然出现。
我的配置：其他配置使用默认值
Properties config = new Properties();
config.put("compiler", "httl.spi.compilers.JavassistCompiler");
config.put("json.codec", "httl.spi.codecs.JacksonCodec");
config.put("cache.capacity", "1000");
config.put("loggers", "httl.spi.loggers.Log4jLogger");
// 测试使用pan
config.put("output.stream", "false");
config.put("loaders+", "httl.spi.loaders.JarLoader");

因httl已用于公司产品，望优先予以关注，十分感激。

**Comment by PanShaobo** [comment]
问题解决了。
错误原因：was重新启动应用不会将动态的template类从javassist的ClassPool中卸载，导致再次创建的$TXT那些静态变量调用CharCache的id值不一致。存储CharCache的id值是重新计算的，但是类却是取出上次缓存在ClassPool中的。
解决方法：在JavassistCompiler中return cls.toClass();前增加了一行cls.detach();使ClassPool不缓存class信息，同事也节省了缓存空间。

测试：在was8.5，weblogic11g，jetty上测试通过
